torch/csrc/utils/python_dispatch.cpp

```diff
@@ -50,6 +53,90 @@ inline torch::CppFunction dispatch_str(const char* key, Func&& raw_f) {
   }
 }
 
+class PythonKernelHolder : public c10::OperatorKernel {
+  SafePyObject* func_;
+public:
+  PythonKernelHolder(py::object func) : func_(new SafePyObject(func.release().ptr(), getPyInterpreter())) {}
+  bool is_valid_call(const c10::impl::PyInterpreter* ip) const {
+    return func_->has_same_interpreter(ip);
+  }
+  // This is a generally useful pattern and safer than directly using pybind11's
+  // py::object destructor.  This is because this object may outlive
+  // libtorch_python, so we want to disarm the deallocation if that happens.
+  // PyInterpreter does this correctly, pybind11 does not.
+  ~PythonKernelHolder() {
+    getPyInterpreter()->decref(func_->ptr(getPyInterpreter()), /*is_tensor*/ false);
+  }
+
+  void operator()(const c10::OperatorHandle& op, c10::DispatchKeySet keyset, torch::jit::Stack* stack) {
+    // If there are multiple python interpreters overriding the same op, then the op registered by the last
+    // interpreter will be available in the dispatch table. This can lead different interpreters to call into
+    // PyObjects from a different python program.
+    // Thankfully, we maintain a list of kernels that were ever registered (and haven't been deregistered yet),
+    // so if the PyInterpreter of func_ doesn't match with the current PyInterpreter, we look up into previously
+    // registered kernels.
+    // This is also useful when one PyInterpreter overrides an op, but another PyInterpreter just wants to call into
+    // the native operation.
+    if (!func_->has_same_interpreter(getPyInterpreter())) {
+      const auto& kernels = op.get_all_saved_kernels(keyset.highestPriorityTypeId());
+      for (const auto& ker: kernels) {
```

Comment:
Overriding:
Two py interpreters: AutoCast (no sum)
my_sum (2)

what happens if we need to fallthrough?